src/Fable.Transforms/FSharp2Fable.Util.fs

```diff
@@ -1363,6 +1363,7 @@ module TypeHelpers =
                 // This is necessary because: `makeType` reduces Erased Unions (including Ux) to `Any` -> no type info any more
                 //
                 // Note: no handling of nested types: `U2<string, U<int, float>>` -> `int` & `float` don't get extract
+                let ty = ty.AbbreviatedType
```

Comment:
There is a `nonAbbreviatedType` helper in FSharp2Fable.Util that deals with edge cases (e.g. alias of an alias) that could be used here. Latest FCS also has a `StripAbbreviations` method, not sure if it can be used.